siglens/siglens
# Favorite Dashboard Functionality (Backend changes)

# Description
This PR introduces the ability to mark dashboards as favorites. It adds a new boolean attribute `isFavorite` to the dashboard JSON, which reflects the favorite status. The favorite status can be toggled via a new API endpoint PUT `/dashboards/favorite/{dashboard-id}`

Fixes #151

# Testing
- Create a dashboard from the UI. The `isFavorite` property is initially set to false.
- Call the PUT `/dashboards/favorite/{dashboard-id}` endpoint with a valid dashboard ID and no request body. The response should return the updated `isFavorite` status.
- Verify the favorite status of the dashboard by inspecting the `isFavorite` attribute, which can be obtained by using the GET `dashboards/{dashboard-id}` API to fetch the dashboard details.
- Toggle the favorite status multiple times and ensure the `isFavorite` status is updated correctly each time.

# Checklist:
Before marking your pull request as ready for review, complete the following.
